src/riverroute/mosart_physics_mod.F90

```diff
@@ -0,0 +1,604 @@
+      end if
+
+   end subroutine mainchannelRouting
+
+   !-----------------------------------------------------------------------
+
+   subroutine Routing_KW(nr, nt, theDeltaT)
+      !  classic kinematic wave routing method
+
+      ! Arguments
+      integer, intent(in) :: nr, nt
+      real(r8), intent(in) :: theDeltaT
+
+      ! Local variables
+      integer  :: k
+      real(r8) :: temp_gwl, temp_dwr, temp_gwl0
+
+      ! estimate the inflow from upstream units
+      TRunoff%erin(nr,nt) = 0._r8
+      TRunoff%erin(nr,nt) = TRunoff%erin(nr,nt) - TRunoff%eroutUp(nr,nt)
+
+      ! estimate the outflow
+      if(TUnit%rlen(nr) <= 0._r8) then ! no river network, no channel routing
+         TRunoff%vr(nr,nt) = 0._r8
+         TRunoff%erout(nr,nt) = -TRunoff%erin(nr,nt)-TRunoff%erlateral(nr,nt)
+      else
+         if(TUnit%areaTotal2(nr)/TUnit%rwidth(nr)/TUnit%rlen(nr) > 1e6_r8) then
+            TRunoff%erout(nr,nt) = -TRunoff%erin(nr,nt)-TRunoff%erlateral(nr,nt)
+         else
+            TRunoff%vr(nr,nt) = CRVRMAN_nosqrt(TUnit%rslpsqrt(nr), TUnit%nr(nr), TRunoff%rr(nr,nt))
+            TRunoff%erout(nr,nt) = -TRunoff%vr(nr,nt) * TRunoff%mr(nr,nt)
+            if(-TRunoff%erout(nr,nt) > TINYVALUE .and. TRunoff%wr(nr,nt) + &
+              (TRunoff%erlateral(nr,nt) + TRunoff%erin(nr,nt) + TRunoff%erout(nr,nt)) * theDeltaT < TINYVALUE) then
+               TRunoff%erout(nr,nt) = &
+                    -(TRunoff%erlateral(nr,nt) + TRunoff%erin(nr,nt) + TRunoff%wr(nr,nt) / theDeltaT)
+               if(TRunoff%mr(nr,nt) > 0._r8) then
+                  TRunoff%vr(nr,nt) = -TRunoff%erout(nr,nt) / TRunoff%mr(nr,nt)
+               end if
+            end if
+         end if
+      end if
+
+      temp_gwl = TRunoff%qgwl(nr,nt) * TUnit%area(nr) * TUnit%frac(nr)
+
+      TRunoff%dwr(nr,nt) = TRunoff%erlateral(nr,nt) + TRunoff%erin(nr,nt) + TRunoff%erout(nr,nt) + temp_gwl
+
+      if((TRunoff%wr(nr,nt)/theDeltaT &
+           + TRunoff%dwr(nr,nt)) < -TINYVALUE) then
+         write(iulog,*) 'mosart: ERROR main channel going negative: ', nr, nt
+         write(iulog,*) theDeltaT, TRunoff%wr(nr,nt), &
+              TRunoff%wr(nr,nt)/theDeltaT, TRunoff%dwr(nr,nt), temp_gwl
+         write(iulog,*) ' '
+      endif
+
+      ! check for stability
+      !    if(TRunoff%vr(nr,nt) < -TINYVALUE .or. TRunoff%vr(nr,nt) > 30) then
+      !       write(iulog,*) "Numerical error inRouting_KW, ", nr,nt,TRunoff%vr(nr,nt)
+      !    end if
+
+      ! check for negative wr
+      !    if(TRunoff%wr(nr,nt) > 1._r8 .and. &
+      !      (TRunoff%wr(nr,nt)/theDeltaT + TRunoff%dwr(nr,nt))/TRunoff%wr(nr,nt) < -TINYVALUE) then
+      !       write(iulog,*) 'negative wr!', TRunoff%wr(nr,nt), TRunoff%dwr(nr,nt), temp_dwr, temp_gwl, temp_gwl0, theDeltaT
+      !       stop
+      !    end if
+
+   end subroutine Routing_KW
+
+   !-----------------------------------------------------------------------
+
+   subroutine Routing_MC(nr, nt, theDeltaT)
```

Comment:
Should these return something about being unsupported?  Or just be removed?

Reply:
address now or open an issue?

Reply:
Please open an issue as well.

Reply:
These actually have already been removed.